msal/build.gradle

```diff
@@ -96,6 +107,52 @@ android {
     }
 }
 
+ext.createCoverageTask = { buildVariant, testType ->
+    def unitTest = "test${buildVariant.capitalize()}UnitTest"
+    def androidTest = "connected${buildVariant.capitalize()}AndroidTest"
+    def unitTestFile = "jacoco/${unitTest}.exec"
+    def androidTestFile = "outputs/code_coverage/${buildVariant}AndroidTest/connected/*.ec"
+    def taskList, executionFiles
+    if (testType.matches("BothTest")) {
+        taskList = [unitTest, androidTest]
+        executionFiles = [unitTestFile, androidTestFile]
+    } else if (testType.matches("UnitTest")){
+        taskList = [unitTest]
+        executionFiles = [unitTestFile]
+    } else {
+        taskList = [androidTest]
+        executionFiles = [androidTestFile]
+    }
+
+    task "${buildVariant}Msal${testType}CoverageReport" (type:JacocoReport, dependsOn: taskList) {
+        group = "Code Coverage"
+        description = "Generate Msal Coverage Reports(${testType}) on the ${buildVariant.capitalize()}"
+        def coverageSourceDirs = "${projectDir}/src/main/java"
+        def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test.*']
+        def javaClasses = fileTree(
+                dir: "${buildDir}/intermediates/javac/${buildVariant}/classes",
+                excludes: fileFilter
+        )
+        //To include common files in the coverage report
+        def commonBuild = buildVariant.matches("(.*)Debug(.*)") ? "debug" : "release"
+        def additionalCoverageSourceDirs = "${projectDir}/../common/common/src/main/java"
+        def additionalJavaClasses = fileTree(
+                dir: "${buildDir}/../common/intermediates/javac/${commonBuild}/classes",
+                excludes: fileFilter
+        )
+
+        sourceDirectories.from = files([coverageSourceDirs])
+        additionalSourceDirs.from = files(additionalCoverageSourceDirs)
+        additionalClassDirs.from = files([additionalJavaClasses])
+        classDirectories.from = files([javaClasses])
+        executionData.from = fileTree(dir: "$buildDir", includes: executionFiles)
+        reports {
+            xml.enabled = true
```

Comment:
can we also get html?

Reply:
html reports are already being generated, by default. So, I didn't add any flag for it